src/main/java/com/somemore/community/service/query/CommunityBoardQueryService.java

```diff
@@ -0,0 +1,82 @@
+import com.somemore.community.usecase.query.CommunityBoardQueryUseCase;
+import com.somemore.global.exception.BadRequestException;
+import com.somemore.volunteer.dto.response.VolunteerForCommunityResponseDto;
+import com.somemore.volunteer.usecase.query.FindVolunteerIdUseCase;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.somemore.global.exception.ExceptionMessage.NOT_EXISTS_COMMUNITY_BOARD;
+
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class CommunityBoardQueryService implements CommunityBoardQueryUseCase {
+
+    private final CommunityBoardRepository communityBoardRepository;
+    private final CenterQueryUseCase centerQueryUseCase;
+    private final FindVolunteerIdUseCase findVolunteerIdUseCase;
+
+    @Override
+    public List<CommunityBoardGetResponseDto> getCommunityBoards() {
+        List<CommunityBoard> boards = communityBoardRepository.getCommunityBoards();
+
+        return boards.stream()
+                .map(board -> {
+                    String writerNickname = getWriterNickname(board.getWriterId());
+                    return CommunityBoardGetResponseDto.fromEntity(board, writerNickname);
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<CommunityBoardGetResponseDto> getCommunityBoardsByWriterId(UUID writerId) {
+        List<CommunityBoard> boards = communityBoardRepository.getCommunityBoardsByWriterId(writerId);
+        String writerNickname = getWriterNickname(writerId);
+
+        return boards.stream()
+                .map(board -> CommunityBoardGetResponseDto.fromEntity(board, writerNickname))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public CommunityBoardGetDetailResponseDto getCommunityBoardDetail(Long id) {
+        CommunityBoard board = communityBoardRepository.findById(id)
+                .orElseThrow(() -> new BadRequestException(NOT_EXISTS_COMMUNITY_BOARD.getMessage()));
+
+        return CommunityBoardGetDetailResponseDto.fromEntity(board, getWriterDetail(board.getWriterId()));
+    }
+
+    private String getWriterNickname(UUID writerId) {
+        String nickname = findVolunteerIdUseCase.getNicknameById(writerId);
+
+        if (nickname == null) {
+            nickname = centerQueryUseCase.getNameById(writerId);
+        }
+
+        return nickname;
+    }
+
+    private WriterDetailDto getWriterDetail(UUID writerId) {
+        VolunteerForCommunityResponseDto volunteer = findVolunteerIdUseCase.getVolunteerDetailForCommunity(writerId);
+
+        if (volunteer == null) {
+            return centerQueryUseCase.getCenterDetailForCommunity(writerId);
+        } else {
+            return volunteer;
+        }
+    }
+}
```

Comment:
이런 문제가 있군요... 찝찝하시겠어요 ㅠ

Reply:
이게 최선일까요..? 마음에 안드네요..

Reply:
해결하실 수 있을거에요. 우선 진행해보시죠!

```suggestion
    private String getWriterNickname(UUID writerId) {
        String nickname = findVolunteerIdUseCase.getNicknameById(writerId);

        if (nickname == null) {
            nickname = centerQueryUseCase.getNameById(writerId);
        }
        return nickname;
    }

    private WriterDetailDto getWriterDetail(UUID writerId) {
        VolunteerForCommunityResponseDto volunteer = findVolunteerIdUseCase.getVolunteerDetailForCommunity(writerId);

        if (volunteer == null) {
            return centerQueryUseCase.getCenterDetailForCommunity(writerId);
        } 
        return volunteer;
        
    }
}
```

Reply:
수정했습니다!